Homepage and footer surface top categories, retire collections (#160)
Swizec/swizec.com: pages/index.mdx

Getting that senior title is easy. Just stick around. Being a true senior takes a new way of thinking. Do you know it?

<LatestArticles pattern="Senior Mindset" limit={3} />

<SmartLink className="button" href="/categories/senior-mindset/">Read more Senior Mindset essays 👉</SmartLink>

### Ship fast without breaking things

> Every team slows down as it grows. More people, more code, more meetings, less shipping. It doesn't have to be that way. These essays are the thinking behind my book [Scaling Fast](https://scalingfastbook.com).

Author: ```suggestion
Every team slows down as it grows. More people, more code, more meetings, less shipping. But you don’t have to slow down. These essays are the thinking behind my book [Scaling Fast: Software Engineering Through the Hockeystick](https://scalingfastbook.com).
```

<LatestArticles pattern="Scaling Fast Book|Velocity|Teamwork|Complexity" limit={3} />

<SmartLink className="button" href="/scaling-fast/">Read more essays on shipping fast 👉</SmartLink>

### Use AI without losing the plot

> AI writes most of my code these days. The hard part was never the typing — it's knowing what to build and recognizing when the machine is wrong. Lessons from using AI in production, not demos.

Author: ```suggestion
Coding got easy with AI but the engineering remains. You have to know what to build, recognize when the machine is wrong, and design systems that work at scale. Here I share lessons from using AI in production.
```

<LatestArticles pattern="\bAI\b" limit={3} />

<SmartLink className="button" href="/categories/ai/">Read more AI essays 👉</SmartLink>

### Read JavaScript and TypeScript lessons from production

> I took a bet on JavaScript in 2005 when it was known as DHTML. It's been my favorite language ever since. Since 2019, I've been full into TypeScript – perfect balance between flexibility and correctness.

Author: ```suggestion
JavaScript has been my programming language of choice since 2005 when it was called DHTML. Since 2019, I'm full into TypeScript — the perfect balance between flexibility and correctness.
```

<LatestArticles pattern="JavaScript|TypeScript" limit={3} />

<SmartLink className="button" href="/categories/javascript/">Read more JavaScript and TypeScript lessons from production 👉</SmartLink>

### Take your React beyond the tutorial

> React is here to stay. It solves many problems you may not even know existed. These essays are the good stuff I learned over the years.

Author: ```suggestion
React is here to stay. It solves problems you may not even know existed. These essays are the good stuff I learned over the years.
```

<LatestArticles pattern="React|Frontend" limit={3} />

<SmartLink className="button" href="/categories/react/">Read more React lessons from production 👉</SmartLink>

### Find insights into Serverless and modern backends

> I've been building web backends since ~2004 when they were just called websites. With these essays I want to share the hard lessons learned.

Author: ```suggestion
I've been building web backends since ~2004 when they were just called websites. With these essays I want to share the hard lessons learned about serverless, databases, APIs, and the ops that keep them alive.
```

<LatestArticles pattern="Serverless|Backend" limit={3} />

<SmartLink className="button" href="/categories/backend/">Read more serverless and backend articles 👉</SmartLink>

### Lead without becoming a manager

Teams, mentoring, feedback, hiring — the people skills that turn a strong engineer into a force multiplier. No MBA required.

<LatestArticles pattern="Leadership" limit={3} />

<SmartLink className="button" href="/categories/leadership/">Read more Leadership essays 👉</SmartLink>

### Grow your engineering career

> Your code gets you to senior. What gets you further is everything around the code — interviews, negotiation, picking the right problems, and knowing what the business actually pays you for.

Author: ```suggestion
Your code gets you to senior. To go further you need more than code — interviews, negotiation, picking the right problems, and knowing what the business actually pays you for. These essays cover the career side of software engineering.
```

<LatestArticles pattern="Career|Interviewing|Salary|Compensation|Hiring" limit={3} />

<SmartLink className="button" href="/career-growth/">Read more career essays 👉</SmartLink>

### Go straight to the source

> Blogs and hot takes are fun but the good stuff hides in primary sources. I read computer science papers and books cover to cover, then write about what holds up in production and what doesn't.

Author: ```suggestion
Blogs and hot takes are fun but the good stuff hides in primary sources. I like to read computer science papers and books cover to cover and write about what holds up in production and what doesn't.
```
